app/src/main/java/fr/neamar/kiss/preference/AddSearchProviderPreference.java

```diff
@@ -154,10 +153,10 @@ private boolean validate() {
             //cancel close dialog
             return false;
         }
-        //check if a valid url is given
-        if (!validateUrl()) {
-            Toast.makeText(this.getContext(), R.string.search_provider_error_url, Toast.LENGTH_SHORT).show();
-            //not a url
+        //check if a valid uri is given
+        if (!validateUri()) {
+            Toast.makeText(this.getContext(), R.string.search_provider_error_uri, Toast.LENGTH_SHORT).show();
```

Comment:
Let `search_provider_error_url` as is and use distinct string for URI check.
Existing string is not always using text "URL", but also "web address", "link" and other synonyms which won't match anymore now.

Reply:
Updated @marunjar !